backend/books/views.py

```diff
@@ -361,14 +396,27 @@ def convert_zero_to_null(self, data):
 
         return data
 
+    def handle_image(self, request, setDefaultImage):
+        has_image_bytes = self.has_image_bytes(request)
+        has_image_url = self.has_image_url(request)
+        return has_image_bytes or has_image_url or setDefaultImage
+    
+    def has_image_bytes(self, request):
+        return request.FILES.get('image_bytes', None) is not None
+
+    def has_image_url(self, request):
+        return request.data.get('image_url', None) is not None
+
     def bookimage_get_and_create(self, request, isbn_13, setDefaultImage):
         book = Book.objects.filter(isbn_13=isbn_13)
 
         # This creates an image in static and sends a file
         if setDefaultImage:
             url = self.isbn_toolbox.get_default_image_url()
-        else:
+        elif self.has_image_bytes(request):
             url = self.isbn_toolbox.commit_image_raw_bytes(request, book[0].id, isbn_13)
+        elif self.has_image_url(request):
+            url = self.isbn_toolbox.commit_image_url(request, book[0].id, isbn_13)
```

Comment:
Same as above with meeting these conditions otherwise `url` is undefined

Reply:
answered above